drivers/sht1x/include/sht1x_types.h

```diff
@@ -0,0 +1,82 @@
+/*
```

Comment:
I don't think its a good idea to cut out elements of the public driver API into its own file. As part of the public API they should be placed in that API file.

Reply:
`drivers/include/sht1x.h` depends on the definitions in `drivers/sht1x/include/sht1x_params.h` to be used. But `sht1x_params.h` needs the types to define the params. The only way to solve this is by splitting the dependency cycle, as done here.